event-svc/src/event/validator/grouped.rs

```diff
@@ -0,0 +1,125 @@
+use ceramic_event::unvalidated::{self, signed};
+use ipld_core::ipld::Ipld;
+
+use super::UnvalidatedEvent;
+
+#[derive(Debug)]
+pub enum ValidationNeeded {
+    SignedInit(SignedInit),
+    SignedData(SignedData),
+    Time(Time),
+    None(Unsigned),
+}
+
+impl From<UnvalidatedEvent> for ValidationNeeded {
+    fn from(value: UnvalidatedEvent) -> Self {
+        match value.event.as_ref() {
+            unvalidated::Event::Time(_) => Self::Time(Time(value)),
+            unvalidated::Event::Signed(signed) => match signed.payload() {
+                unvalidated::Payload::Data(_) => Self::SignedData(SignedData(value)),
+                unvalidated::Payload::Init(_) => Self::SignedInit(SignedInit(value)),
+            },
+            unvalidated::Event::Unsigned(_) => Self::None(Unsigned(value)),
+        }
+    }
+}
+
+macro_rules! unvalidated_wrapper {
```

Comment:
but this is "simple macros" ;) i'll try to clarify this file, it was supposed to be magic/internal helpers to remove the need to match on event types and make it impossible to represent them wrong. That is, instead of adding `unvalidated::Event<D>`  that we know is a time event and then having to do the match and panic on the other branches, this takes care of hiding that so when we build it and it declares itself a time event, we can just call `as_time()` and get what we want.